### PR TITLE
Simplify logic in p::d::Triangulation.

### PR DESCRIPTION
**Pull request by bangerth**
The logic in p::d::T::execute_coarsening_and_refinement() is overly complicated
because the function calls itself recursively (indirectly, on top of that). But
this is not necessary if the place where that happens just calls the function
in the base class -- which happens to be easy to do by explicitly qualifying
the function call.

Do so.

@tjhei?

### CODE REVIEW

File: source/distributed/tria.cc [hunk 1]
```diff
@@ -2677,21 +2670,16 @@ namespace parallel
 
           // actually do the refinement but prevent the refinement hook below
           // from taking over
```
Comment (tjhei) [inline source/distributed/tria.cc]: can you replace this comment please? Something like:
"execute the refinement to change the local mesh by calling the base class refinement function directly"

**Comment by tjhei** [comment]
I like it (except that I need to rebase my branch now). Grmbl.

**Comment by bangerth** [comment]
I've changed the comment as suggested. Sorry to step on your turf -- but we talked about ideas how you can do this going forward with less pain.

**Comment by tjhei** [comment]
> but we talked about ideas how you can do this going forward with less pain.

Yes, I know. ;-)

**Pull request merged**